tests/test_latentdistributiontest.py

```diff
@@ -56,13 +58,76 @@ def test_bad_kwargs(self):
             LatentDistributionTest(size_correction=0)
         with self.assertRaises(TypeError):
             LatentDistributionTest(pooled=0)
+        with self.assertRaises(TypeError):
+            ldt = LatentDistributionTest(input_graph="hello")
 
     def test_n_bootstraps(self):
         for test in self.tests.keys():
             ldt = LatentDistributionTest(test, self.tests[test], n_bootstraps=123)
             ldt.fit(self.A1, self.A2)
             self.assertEqual(ldt.null_distribution_.shape[0], 123)
 
+    def test_passing_networkx(self):
+        np.random.seed(123)
+        A1 = er_np(20, 0.8)
+        A2 = er_np(20, 0.8)
+        A1_nx = nx.from_numpy_matrix(A1)
+        A2_nx = nx.from_numpy_matrix(A2)
+        # check passing nx, when exepect embeddings
+        with self.assertRaises(TypeError):
+            ldt = LatentDistributionTest(input_graph=False)
+            ldt.fit_predict(A1_nx, A2)
+        with self.assertRaises(TypeError):
+            ldt = LatentDistributionTest(input_graph=False)
+            ldt.fit_predict(A1, A2_nx)
+        with self.assertRaises(TypeError):
+            ldt = LatentDistributionTest(input_graph=False)
+            ldt.fit_predict(A1_nx, A2_nx)
+        # check that the appropriate input works
+        ldt = LatentDistributionTest(input_graph=True)
+        ldt.fit_predict(A1_nx, A2_nx)
+
+    def test_passing_embeddings(self):
+        np.random.seed(123)
+        A1 = er_np(20, 0.8)
+        A2 = er_np(20, 0.8)
+        ase_1 = AdjacencySpectralEmbed(n_components=2)
+        X1 = ase_1.fit_transform(A1)
+        ase_2 = AdjacencySpectralEmbed(n_components=2)
+        X2 = ase_2.fit_transform(A2)
+        ase_3 = AdjacencySpectralEmbed(n_components=1)
+        X3 = ase_3.fit_transform(A2)
+        # check embeddings having weird ndim
+        with self.assertRaises(ValueError):
+            ldt = LatentDistributionTest(input_graph=False)
+            ldt.fit_predict(X1, X2.reshape(-1, 1, 1))
+        with self.assertRaises(ValueError):
+            ldt = LatentDistributionTest(input_graph=False)
+            ldt.fit_predict(X1.reshape(-1, 1, 1), X2)
+        # check embeddings having mismatching number of components
+        with self.assertRaises(ValueError):
+            ldt = LatentDistributionTest(input_graph=False)
+            ldt.fit_predict(X1, X3)
+        with self.assertRaises(ValueError):
+            ldt = LatentDistributionTest(input_graph=False)
+            ldt.fit_predict(X3, X1)
+        # check passing weird stuff as input (caught by check_array)
+        with self.assertRaises(TypeError):
```

Comment:
```from graspy.inference import LatentDistributionTest

ldt = LatentDistributionTest(input_graph=False)


class Bob:
    pass


ldt.fit(Bob(), Bob())
```

`TypeError: float() argument must be a string or a number, not 'Bob'`

Reply:
while this is a type error being raised by sklearn, not quite what we want cause it's not descriptive enough, methinks. See my other comment about checking to make sure it's an array explicitly prior to this.

Reply:
resolved by checking for ndarray specifically.

```
>>> from graspy.inference import LatentDistributionTest
>>> ldt = LatentDistributionTest(input_graph=False)
>>> 
>>> 
>>> class Bob:
...     pass
... 
>>> 
>>> ldt.fit(Bob(), Bob())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/antonalyakin/Documents/Code/graspy/graspy/inference/latent_distribution_test.py", line 344, in fit
    raise TypeError(msg)
TypeError: embedding of the first graph is an <class '__main__.Bob'>, not an array! if input_graph is False, the inputs need to be adjacency spectral embeddings, with shapes (n, d) and (m, d), passed as np.ndarrays!
```